allauth/socialaccount/helpers.py

```diff
@@ -20,40 +20,39 @@
 
 
 def _process_signup(request, sociallogin):
-    auto_signup = get_adapter(request).is_auto_signup_allowed(
-        request,
-        sociallogin)
+    social_adapter = get_adapter(request)
+    auto_signup = social_adapter.is_auto_signup_allowed(request, sociallogin)
     if not auto_signup:
         request.session['socialaccount_sociallogin'] = sociallogin.serialize()
         url = reverse('socialaccount_signup')
-        ret = HttpResponseRedirect(url)
-    else:
-        # Ok, auto signup it is, at least the e-mail address is ok.
-        # We still need to check the username though...
-        if account_settings.USER_MODEL_USERNAME_FIELD:
-            username = user_username(sociallogin.user)
-            try:
-                get_account_adapter().clean_username(username)
-            except ValidationError:
-                # This username is no good ...
-                user_username(sociallogin.user, '')
-        # FIXME: This part contains a lot of duplication of logic
-        # ("closed" rendering, create user, send email, in active
-        # etc..)
+        return HttpResponseRedirect(url)
```

Comment:
Can I introduce you to the wonderful, wonderful `q` library? :)

https://pypi.python.org/pypi/q